src/Shouldly/MessageGenerators/ShouldBeSubsetOfMessageGenerator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Shouldly.MessageGenerators
+{
+    internal class ShouldBeSubsetOfMessageGenerator : ShouldlyMessageGenerator
+    {
+        private static readonly Regex Validator = new Regex("ShouldBeSubsetOf", RegexOptions.Compiled);
+
+        public override bool CanProcess(ShouldlyAssertionContext context)
+        {
+            return Validator.IsMatch(context.ShouldMethod);
+        }
+
+        public override string GenerateErrorMessage(ShouldlyAssertionContext context)
+        {
+            const string format = @"{0} should be subset of {1} but {2} do not";
```

Comment:
Maybe `{0} should be subset of {1} but {2} are outside subset` as it is quite explicit. 

Then this is getting really nitpicky but a single item should read `is` not `are`. For example

 - `new[] { 1, 2, 5 } should be subset of  [2, 3, 4] but [1, 5] are outside subset`
 - `new[] { 2, 3, 5 } should be subset of  [2, 3, 4] but [1] is outside subset`